core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -605,6 +645,77 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord]) extends PSBTMap {
       .filter(element => PSBTInputKeyId.fromByte(element.key.head) == key)
       .asInstanceOf[Vector[T]]
   }
+
+  def isFinalized: Boolean =
+    getRecords(FinalizedScriptSigKeyId).nonEmpty || getRecords(
+      FinalizedScriptWitnessKeyId).nonEmpty
+
+  /**
+    * Takes the InputPSBTMap returns a UTXOSpendingInfo
+    * that can be used to sign the input
+    * @param txIn The transaction input that this InputPSBTMap represents
+    * @param signers Signers that will be used to sign the input
+    * @return A corresponding UTXOSpendingInfo
+    */
+  def toUTXOSpendingInfo(
+      txIn: TransactionInput,
+      signers: Seq[Sign]): UTXOSpendingInfo = {
+    toUTXOSpendingInfo(txIn, signers, ConditionalPath.NoConditionsLeft)
+  }
+
+  /**
+    * Takes the InputPSBTMap returns a UTXOSpendingInfo
+    * that can be used to sign the input
+    * @param txIn The transaction input that this InputPSBTMap represents
+    * @param signers Signers that will be used to sign the input
+    * @param conditionalPath Path that should be used for the script
+    * @return A corresponding UTXOSpendingInfo
+    */
+  def toUTXOSpendingInfo(
+      txIn: TransactionInput,
+      signers: Seq[Sign],
+      conditionalPath: ConditionalPath): UTXOSpendingInfo = {
+    require(!isFinalized, s"Cannot update an InputPSBTMap that is finalized")
+
+    val outPoint = txIn.previousOutput
+
+    val witVec = getRecords[WitnessUTXO](WitnessUTXOKeyId)
+    val txVec = getRecords[NonWitnessOrUnknownUTXO](NonWitnessUTXOKeyId)
+
+    val output = if (witVec.size == 1) {
+      witVec.head.spentWitnessTransaction
+    } else if (txVec.size == 1) {
+      val tx = txVec.head.transactionSpent
+      tx.outputs(txIn.previousOutput.vout.toInt)
+    } else {
+      throw new UnsupportedOperationException(
+        "Not enough information in the InputPSBTMap to get a valid UTXOSpendingInfo")
+    }
+
+    val redeemScriptVec = getRecords[RedeemScript](RedeemScriptKeyId)
+    val redeemScriptOpt =
+      if (redeemScriptVec.size == 1) Some(redeemScriptVec.head.redeemScript)
+      else None
+
+    val scriptWitnessVec = getRecords[WitnessScript](WitnessScriptKeyId)
+    val scriptWitnessOpt =
+      if (scriptWitnessVec.size == 1)
+        Some(P2WSHWitnessV0(scriptWitnessVec.head.witnessScript))
```

Comment:
Will this cause issues if the input is a P2WPHK?

Reply:
Yeah, I think this will be an issue. Not really sure what goes in `WitnessScript` in the `P2WPKH` case though, since the witness is not a `ScriptPubKey` but instead is just a PubKey

Reply:
Also, looking at `WitnessScript` now, that `ScriptPubKey` parameter should probably be a `RawScriptPubKey`, doesn't have to happen on this PR though

Reply:
Hopefully my comment above will help resolve this